core/store/src/db.rs

```diff
@@ -145,11 +147,46 @@ impl std::fmt::Display for DBCol {
             Self::ColChunkHashesByHeight => "chunk hashes indexed by height_created",
             Self::ColBlockOrdinal => "block ordinal",
             Self::ColGCCount => "gc count",
+            Self::ColOutcomesByBlockHash => "outcomes by block hash",
         };
         write!(formatter, "{}", desc)
     }
 }
 
+// List of columns for which GC should be implemented
+lazy_static! {
+    pub static ref SHOULD_COL_GC: Vec<bool> = {
+        let mut col_gc = vec![true; NUM_COLS];
+        col_gc[DBCol::ColDbVersion as usize] = false; // DB version is unrelated to GC
+        col_gc[DBCol::ColBlockMisc as usize] = false;
+        col_gc[DBCol::ColBlockHeader as usize] = false; // header sync needs headers
+        col_gc[DBCol::ColGCCount as usize] = false; // GC count it self isn't GCed
+        col_gc[DBCol::ColBlockHeight as usize] = false; // block sync needs it + genesis should be accessible
+        col_gc[DBCol::ColPeers as usize] = false; // Peers is unrelated to GC
+        col_gc[DBCol::ColBlockMerkleTree as usize] = false;
+        col_gc[DBCol::ColAccountAnnouncements as usize] = false;
+        col_gc[DBCol::ColEpochLightClientBlocks as usize] = false;
+        col_gc[DBCol::ColPeerComponent as usize] = false; // Peer related info doesn't GC
+        col_gc[DBCol::ColLastComponentNonce as usize] = false;
+        col_gc[DBCol::ColComponentEdges as usize] = false;
+        col_gc[DBCol::ColBlockOrdinal as usize] = false;
+        col_gc
+    };
+}
+
+// List of columns for which GC may not be executed even in fully operational node
```

Comment:
why are they not executed?

Reply:
I think about it in the following terms:
1. Column should implement GC and will use it.
2. Column should implement GC and it may not use it until specific circumstances.
3. Column should not be GCed.

Columns that are in SHOULD_COL_GC should implement their GC method anyway. However, it doesn't mean that it will be called in any predictable time in fully operational node. For example, with no State Sync called ColStateHeaders will be never cleared because they will not be created. That's why SKIP_COL_GC comes.